kibana/Dockerfile

```diff
@@ -113,16 +113,16 @@ LABEL org.label-schema.build-date="2022-01-31T21:13:59.193Z" \
   org.label-schema.vcs-ref="57ca5e139a33dd2eed927ce98d8231a1f217cd15" \
   org.label-schema.vcs-url="https://github.com/elastic/kibana" \
   org.label-schema.vendor="Elastic" \
-  org.label-schema.version="8.0.0-rc2" \
-  org.opencontainers.image.created="2022-01-31T21:13:59.193Z" \
+  org.label-schema.version="8.0.0" \
+  org.opencontainers.image.created="2022-02-03T17:05:17.654Z" \
```

Comment:
Looks like there's a minor discrepancy here between the pushed images and the committed values again: :see_no_evil:

```diff
-LABEL org.label-schema.build-date=2022-02-03T23:06:21.184Z org.label-schema.license=Elastic License org.label-schema.name=Kibana org.label-schema.schema-version=1.0 org.label-schema.url=https://www.elastic.co/products/kibana org.label-schema.usage=https://www.elastic.co/guide/en/kibana/reference/index.html org.label-schema.vcs-ref=57ca5e139a33dd2eed927ce98d8231a1f217cd15 org.label-schema.vcs-url=https://github.com/elastic/kibana org.label-schema.vendor=Elastic org.label-schema.version=8.0.0 org.opencontainers.image.created=2022-02-03T23:06:21.184Z org.opencontainers.image.documentation=https://www.elastic.co/guide/en/kibana/reference/index.html org.opencontainers.image.licenses=Elastic License org.opencontainers.image.revision=57ca5e139a33dd2eed927ce98d8231a1f217cd15 org.opencontainers.image.source=https://github.com/elastic/kibana org.opencontainers.image.title=Kibana org.opencontainers.image.url=https://www.elastic.co/products/kibana org.opencontainers.image.vendor=Elastic org.opencontainers.image.version=8.0.0
+LABEL org.label-schema.build-date=2022-02-03T17:05:17.654Z org.label-schema.license=Elastic License org.label-schema.name=Kibana org.label-schema.schema-version=1.0 org.label-schema.url=https://www.elastic.co/products/kibana org.label-schema.usage=https://www.elastic.co/guide/en/kibana/reference/index.html org.label-schema.vcs-ref=57ca5e139a33dd2eed927ce98d8231a1f217cd15 org.label-schema.vcs-url=https://github.com/elastic/kibana org.label-schema.vendor=Elastic org.label-schema.version=8.0.0 org.opencontainers.image.created=2022-02-03T17:05:17.654Z org.opencontainers.image.documentation=https://www.elastic.co/guide/en/kibana/reference/index.html org.opencontainers.image.licenses=Elastic License org.opencontainers.image.revision=57ca5e139a33dd2eed927ce98d8231a1f217cd15 org.opencontainers.image.source=https://github.com/elastic/kibana org.opencontainers.image.title=Kibana org.opencontainers.image.url=https://www.elastic.co/products/kibana org.opencontainers.image.vendor=Elastic org.opencontainers.image.version=8.0.0
```

Reply:
> 2022-02-03T23:06:21.184Z

I am not sure where this value is coming from?

Reply:
That's from the image that got pushed to `docker.elastic.co/kibana/kibana:8.0.0@sha256:245736e3b530b153a51512141821cea5f7741df616818a6e8266c26d841be61c`:

```json
$ docker run --rm gcr.io/go-containerregistry/crane config docker.elastic.co/kibana/kibana:8.0.0@sha256:245736e3b530b153a51512141821cea5f7741df616818a6e8266c26d841be61c | jq '.config.Labels'
{
  "org.label-schema.build-date": "2022-02-03T23:06:21.184Z",
  "org.label-schema.license": "Elastic License",
  "org.label-schema.name": "Kibana",
  "org.label-schema.schema-version": "1.0",
  "org.label-schema.url": "https://www.elastic.co/products/kibana",
  "org.label-schema.usage": "https://www.elastic.co/guide/en/kibana/reference/index.html",
  "org.label-schema.vcs-ref": "57ca5e139a33dd2eed927ce98d8231a1f217cd15",
  "org.label-schema.vcs-url": "https://github.com/elastic/kibana",
  "org.label-schema.vendor": "Elastic",
  "org.label-schema.version": "8.0.0",
  "org.opencontainers.image.created": "2022-02-03T23:06:21.184Z",
  "org.opencontainers.image.documentation": "https://www.elastic.co/guide/en/kibana/reference/index.html",
  "org.opencontainers.image.licenses": "Elastic License",
  "org.opencontainers.image.revision": "57ca5e139a33dd2eed927ce98d8231a1f217cd15",
  "org.opencontainers.image.source": "https://github.com/elastic/kibana",
  "org.opencontainers.image.title": "Kibana",
  "org.opencontainers.image.url": "https://www.elastic.co/products/kibana",
  "org.opencontainers.image.vendor": "Elastic",
  "org.opencontainers.image.version": "8.0.0"
}
```

Reply:
OK, thanks for the clarification.

Let me explain how our workflow works because I am not sure these values have been in sync for the other release versions as well.

Each Elastic project has its own build workflow to build their docker artifacts (e.g. ES uses Gradle, Logstash uses rake, Kibana uses node, Elastic Agent uses mage and so on).
We do have a unified release process where we ask each team to build:

1. docker images for amd64 and arm64 (we have been doing it for years)
2. docker build context (for Docker library, the DoD workflow and so on)

Not all the teams are using `Dockerfile` files to build their docker images in their build workflow. But, even if they are not, we do ask them to provide a "docker build context" artifact for (2).
All these artifacts are saved as tarball files in our source of truth bucket and we do have an internal mechanism to identify the release metadata for all these artifacts.

As an example, here are the docker artifacts for Kibana (+ checksums and gpg signatures):

```
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-build-context.tar.gz
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-build-context.tar.gz.sha512
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-build-context.tar.gz.asc

- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-image-aarch64.tar.gz
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-image-aarch64.tar.gz.sha512
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-image-aarch64.tar.gz.asc

- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-image.tar.gz
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-image.tar.gz.sha512
- https://artifacts.elastic.co/downloads/kibana/kibana-8.0.0-docker-image.tar.gz.asc

- https://artifacts.elastic.co/downloads/kibana/kibana-ironbank-8.0.0-docker-build-context.tar.gz
- https://artifacts.elastic.co/downloads/kibana/kibana-ironbank-8.0.0-docker-build-context.tar.gz.sha512
- https://artifacts.elastic.co/downloads/kibana/kibana-ironbank-8.0.0-docker-build-context.tar.gz.asc

- https://artifacts.elastic.co/downloads/kibana/kibana-ubi8-8.0.0-docker-build-context.tar.gz
- https://artifacts.elastic.co/downloads/kibana/kibana-ubi8-8.0.0-docker-build-context.tar.gz.sha512
- https://artifacts.elastic.co/downloads/kibana/kibana-ubi8-8.0.0-docker-build-context.tar.gz.asc

- https://artifacts.elastic.co/downloads/kibana/kibana-ubi8-8.0.0-docker-image.tar.gz
- https://artifacts.elastic.co/downloads/kibana/kibana-ubi8-8.0.0-docker-image.tar.gz.sha512
- https://artifacts.elastic.co/downloads/kibana/kibana-ubi8-8.0.0-docker-image.tar.gz.asc
```

Then we have a dedicated process for pushing all these images to the different registries (our Elastic registry, Docker Hub, Red Hat, AWS ECR, and so on).

That's why in that case the images under `docker.io/elastic` and `docker.elastic.co` are pushed from the `docker-image.tar.gz` tarball files while the PRs in this repository are created from the `docker-build-context.tar.gz` artifacts.

Therefore that does not mean the build-context has been used to create the docker image but more to provide the artifacts needed by vendors. We ensure all these images are created from the same git commit sha1 though.

Let me know if that helps.

Reply:
That's useful context, but doesn't explain how these timestamps have been the same for all the previous images; I've got a CI job that verifies that an image built from these `Dockerfile`s is the same minus very minor differences to what was pushed, and in my own memory I think they've only differed once :sweat_smile:.

I suppose if the next build has the same problem I'll look at adjusting the CI job to also ignore that label as a minor difference. :+1: